src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Orchestrator/TimerAnalyzer.cs

```diff
@@ -21,56 +21,53 @@ public class TimerAnalyzer : DiagnosticAnalyzer
         private static readonly LocalizableString Description = new LocalizableResourceString(nameof(Resources.DeterministicAnalyzerDescription), Resources.ResourceManager, typeof(Resources));
         private const string Category = SupportedCategories.Orchestrator;
         public const DiagnosticSeverity severity = DiagnosticSeverity.Warning;
-        
-        private static DiagnosticDescriptor V1Rule = new DiagnosticDescriptor(DiagnosticId, Title, V2MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
-        private static DiagnosticDescriptor V2Rule = new DiagnosticDescriptor(DiagnosticId, Title, V2MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
 
-        private static DurableVersion version;
+        public static DiagnosticDescriptor V1Rule = new DiagnosticDescriptor(DiagnosticId, Title, V2MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
+        public static DiagnosticDescriptor V2Rule = new DiagnosticDescriptor(DiagnosticId, Title, V2MessageFormat, Category, severity, isEnabledByDefault: true, description: Description);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(V2Rule, V1Rule); } }
+        private static DurableVersion version;
 
-        public override void Initialize(AnalysisContext context)
+        internal static bool RegisterDiagnostic(SyntaxNode method, CompilationAnalysisContext context, SemanticModel semanticModel)
         {
-            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-            context.EnableConcurrentExecution();
-            context.RegisterSyntaxNodeAction(AnalyzeIdentifierTask, SyntaxKind.IdentifierName);
-            context.RegisterSyntaxNodeAction(AnalyzeIdentifierThread, SyntaxKind.IdentifierName);
+            return (AnalyzeIdentifierTask(method, context, semanticModel) |
```

Comment:
Same comment about adding a code comment describing your intentional use of the `|` operator here.